sys/net/gnrc/netif/gnrc_netif_lora.c

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2017 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ *
+ * @}
+ */
+
+#include "assert.h"
+#include "net/gnrc/netif/lora.h"
+#include "net/lora.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#ifdef MODULE_NETDEV_LORA
+
+static int _send(gnrc_netif_t *netif, gnrc_pktsnip_t *pkt);
+static gnrc_pktsnip_t *_recv(gnrc_netif_t *netif);
+
+static const gnrc_netif_ops_t lora_ops = {
+    .send = _send,
+    .recv = _recv,
+    .get = gnrc_netif_get_from_netdev,
+    .set = gnrc_netif_set_from_netdev,
+};
+
+gnrc_netif_t *gnrc_netif_lora_create(char *stack, int stacksize, char priority,
+                                     char *name, netdev_t *dev)
+{
+    return gnrc_netif_create(stack, stacksize, priority, name, dev, &lora_ops);
+}
+
+static gnrc_pktsnip_t *_recv(gnrc_netif_t *netif)
+{
+    netdev_t *dev = netif->dev;
+
+    int pktlen;
+    gnrc_pktsnip_t *payload;
+
+    assert(dev);
+
+    /* see how much data there is to process */
+    pktlen = dev->driver->recv(dev, NULL, 0, NULL);
+    if (pktlen <= 0) {
+        DEBUG("[lora-gnrc] recv: no data available to process\n");
+        return NULL;
+    }
+
+    /* allocate space for the packet in the pktbuf */
+    payload = gnrc_pktbuf_add(NULL, NULL, pktlen, GNRC_NETTYPE_UNDEF);
+    if (payload == NULL) {
+        DEBUG("[lora-gnrc] recv: unable to allocate space in the pktbuf\n");
+        /* tell the driver to drop the packet */
+        dev->driver->recv(dev, NULL, 1, NULL);
+        return NULL;
+    }
+
+    /* copy the complete payload into the packet buffer */
+    dev->driver->recv(dev, payload->data, pktlen, NULL);
```

Comment:
Doesn't LoRA has any L2 headers oO (if so, I guess you *can* use `gnrc_netif2_raw` after all)?

Reply:
(but from what I have seen on some websites, there is a LoRA MHDR)

Reply:
LoRa (as used here) is just radio data. LoRaMac has L2 headers.

Reply:
Ok, now I need some clarification: Where in the LoRa-Stack is this here to go? Is this just some proof-of-concept that will be insignificant as soon as the real LoRaMAC `gnrc_netif` layer comes? Is this supposed to go below LoRaMac (which would be possible with GNRC, but why have the extra thread then)? In either case: I am struggling to see the long-term benefit of this PR, but maybe I'm missing something.

Reply:
> Is this just some proof-of-concept that will be insignificant as soon as the real LoRaMAC gnrc_netif layer comes?

I was more thinking of another netif type for loramac.

> In either case: I am struggling to see the long-term benefit of this PR, but maybe I'm missing something.

Or maybe I am missing something (most likely I would say ;) )

Reply:
@aabadie wouldn't `gnrc_netif_raw` fit your needs?

Reply:
You actually used [`gnrc_netif_raw`](https://github.com/RIOT-OS/RIOT/pull/8159/files#diff-45bbe6e7306e5287cbc78b86d405120aR59) in #8159 right?

Reply:
@PeterKietzmann, yes, so I think we can close this one.